### PR TITLE
MosaicId and MosaicNonce usability improvement

### PR DESCRIPTION
**Pull request by evias**
## Added 

- Added MosaicNonce unit tests 
- Added MosaicNonce.createRandom() 
- Added MosaicId.createRandom(owner)
- Added MosaicId.createFromNonce(nonce, owner)

### CODE REVIEW

File: src/model/mosaic/MosaicId.ts [hunk 1]
```diff
@@ -33,6 +41,31 @@ export class MosaicId {
      */
     public readonly fullName?: string;
 
+    /**
+     * Create a random MosaicId for given `owner` PublicAccount.
+     * 
+     * @param   owner   {PublicAccount}
+     * @return  {MosaicId}
+     */
+    public static createRandom(owner: PublicAccount): MosaicId {
```
Comment (dgarcia360) [inline src/model/mosaic/MosaicId.ts]: Remove ``createRandom(owner: PublicAccount)`` from mosaicId. The nonce can't be retrieved from the MosaicDefinitionTransaction.

Comment (evias) [reply src/model/mosaic/MosaicId.ts]: Fixed with: https://github.com/nemtech/nem2-sdk-typescript-javascript/pull/68/commits/24fba0b30f6f98b34d3e02477e092cc20ecd33b9

File: src/model/mosaic/MosaicId.ts [hunk 2]
```diff
@@ -33,6 +41,18 @@ export class MosaicId {
...
+     * @param   owner   {Account}
+     * @return  {MosaicId}
+     */
+    public static createFromNonce(nonce: MosaicNonce, owner: PublicAccount): MosaicId {
+        const mosaicId = MosaicIdentifierGenerator(nonce.nonce, convert.hexToUint8(owner.publicKey));
+        return new MosaicId(mosaicId);
+    }
+
     /**
      * Create MosaicId from mosaic and namespace string id (ex: nem:xem or domain.subdom.subdome:token)
```
Comment (dgarcia360) [inline src/model/mosaic/MosaicId.ts]: Update the comment. Search if mosaic ids with the form "namespaceName:mosaicName" appear in other files.

File: src/model/mosaic/MosaicId.ts [hunk 3]
```diff
@@ -43,20 +63,6 @@ export class MosaicId {
         if (id instanceof Array) {
```
Comment (dgarcia360) [inline src/model/mosaic/MosaicId.ts]: is the id: string still required?

File: src/model/mosaic/MosaicId.ts [hunk 4]
```diff
@@ -33,6 +41,18 @@ export class MosaicId {
      */
     public readonly fullName?: string;
```
[the remark this answers is not in the document]
Comment (evias) [reply src/model/mosaic/MosaicId.ts]: not anymore, good point :+1:

File: src/model/mosaic/MosaicId.ts (hunk 2 above)
Comment (evias) [reply src/model/mosaic/MosaicId.ts]: Updated in https://github.com/nemtech/nem2-sdk-typescript-javascript/pull/68/commits/d64579052c78767c3db04cb5464860afe657aae0

File: src/model/mosaic/MosaicId.ts (hunk 3 above)
Comment (evias) [reply src/model/mosaic/MosaicId.ts]: As discussed in private - `string` capacity allows for hexadecimal notation initialization :+1:

File: src/model/mosaic/MosaicId.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (evias) [reply src/model/mosaic/MosaicId.ts]: Fixed in  https://github.com/nemtech/nem2-sdk-typescript-javascript/pull/68/commits/d64579052c78767c3db04cb5464860afe657aae0

**Pull request merged**